docs/docs/introduction/container-images.md

```diff
@@ -0,0 +1,199 @@
+{: .no_toc }
+
+It used to be the case that to develop a simple model - view - control application with some application, database, and web server, you would need to install and run the components on your host. It would be very challenging to work on multiple projects at once. Containers are essential for developers because 
+they allow us to spin up an entire set of components for a specific development environment, and then bring it
+down just as easily.
+
+#### Continuous Integration
+{: .no_toc }
+
+The same is true for testing, You can easily test across operating systems without leaving a single host.
+
+#### Flexible
+{: .no_toc }
+
+The technologies that we use must be flexible to change. The resources
+that we invest in are ephemeral, and change is unavoidable. For a concrete example. let's say you are running
+a complex (containerized) application on a cloud provider. The costs of the provider changes, and you need to do a quick switch! Since your application is containerized, you can easily change providers. You can
+swap out a component and not run things when you don't need to. You aren't forced becoming locked into a cloud vendor. You are a free container whale!
+
+#### Isolation
+{: .no_toc }
+
+While the level of isolation can vary based on the container technology, containers generally allow you
+to isolate an application or component from your host. This is incredibly useful if you have packages
+with conflicting dependencies that need to run on the same host. 
+
+## Virtualization
+{: .no_toc }
+
+You might be familiar with the idea of a [virtual machine](https://en.wikipedia.org/wiki/Virtual_machine) (VM), which was, for some of us, our first foray into virtualization. In a bucket:
+
+ - A hypervisor is based on emulating hardware. 
+ - When we emulate hardware, this is called virtualization.
+
+While the term "virtual machine" is used in different contexts and can spawn much disagreement about
+its definition, for the time being let's abstractly talk about the differences between *containers* and
+*hypervisors*. The biggest difference was already stated - **hypervisor emulation is based on emulating hardware**. With this kind of technology, we take a machine and it emulates virtual hardware, and then bring up 
+another operating system on top of that hardware. There is a nice slide to show [hypervisor based virutualization here](https://www.slideshare.net/Docker/introduction-to-docker-2017/18).
+
+Containers are about virtualizing the subsystems of the operating system itself. We take services that the operating system provides (networking, filesystem, etc.) and provide them in a way that is virtuaizable. This means that I can bring up different copies of an operating system on the same kernel. We are using features of the kernel. So you can think containers as  building on the level of the kernel, and this is different from hypervisors that build one level down, and try to emulate the hardware.
```

Comment:
/s/think containers/think of containers/
/s/building on/running at/
/s/build one/run one/

Reply:
All fixed.